library/Fission/Storage/Mutation/Class.hs

```diff
@@ -0,0 +1,19 @@
+module Fission.Storage.Mutation.Class (MonadDBMutation (..)) where
+
+import           Network.IPFS.CID.Types
+import           Database.Esqueleto hiding ((=.), update)
+
+import           Fission.Models
+import           Fission.Prelude hiding (Value)
+import           Fission.Types
+
+deleteWhereGeneric whereClause = delete <| from \model -> where_ (whereClause model)
+
+class MonadDB m => MonadDBMutation model m where -- Note that this is Monad not MonadDB
+  deleteWhere :: (SqlExpr (Entity model) -> SqlExpr (Value Bool)) -> Transaction m ()
+
+instance MonadDBMutation User Fission where
+  deleteWhere = deleteWhereGeneric
```

Comment:
You can stick this right in the default implementation, too. Then you don't need to write special instances 👍

Reply:
Okay, so these need some tweaks to allow that to happen, but this is all still good:

```haskell
class 
  ( MonadIO m
  , PersistEntity model
  , BackendCompatible SqlBackend (PersistEntityBackend model)
  ) 
  => MonadDBMutation model m where
    deleteWhere :: (SqlExpr (Entity model) -> SqlExpr (Value Bool)) -> Transaction m 
    deleteWhere whereClause = delete <| from \model -> where_ (whereClause model)

class 
  ( MonadIO m
  , PersistEntity model
  , BackendCompatible SqlBackend (PersistEntityBackend model)
  ) 
  => MonadDBQuery model m where
    getBy :: (SqlExpr (Entity model) -> SqlExpr (Value Bool)) -> Transaction m 
    getBy whereClause =
      select <| from \model -> do
        where_ (whereClause model)
        return model
```

Reply:
This now says "any model (that has an `instance PersistModel model`, which they all do) can be used with this interface _automatically_ with this default function. If you want to overwrite it (e.g. in tests) then write an `instance`, but otherwise don't worry about it".